include/xlnt/utils/numeric.hpp

```diff
@@ -144,29 +146,43 @@ class number_serialiser
         return std::string(buf, static_cast<size_t>(len));
     }
 
-    double deserialise(std::string &s) const noexcept
+    // replacement for std::to_string / s*printf("%f", ...)
+    // behaves same irrespective of locale
+    std::string serialise_short(double d) const
     {
-        assert(!s.empty());
+        char buf[30];
+        int len = snprintf(buf, sizeof(buf), "%f", d);
         if (should_convert_comma)
         {
-            // s.data() doesn't have a non-const overload until c++17, hence this little dance
-            convert_pt_to_comma(&s[0], s.size());
+            convert_comma_to_pt(buf, len);
         }
-        return strtod(s.c_str(), nullptr);
+        return std::string(buf, static_cast<size_t>(len));
     }
 
-    double deserialise(const std::string &s) const
+    double deserialise(const std::string &s, ptrdiff_t *len_converted) const
```

Comment:
Change here allows deserialise to be used anywhere std::strtod was used. Matching the char** param is however not possible due to the temporary buffer used when locale conversion is required. In the one case this was required for, the number coverted was what was being checked anyway.